src/ape/contracts/base.py

```diff
@@ -381,13 +381,78 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+    def query(
+        self,
+        start_block: int,
+        stop_block: Optional[int] = None,
+        search_topics: Optional[Dict[str, Any]] = None,
+        extra_addresses: Optional[List] = None,
```

Comment:
Hmm, was this in Banteg's version? Not sure I like this, would rather just compose multiple `.query` outputs